### ISSUE TITLE
adb server version (39) doesn'

### ISSUE THREAD

**Issue by eater** [Genymobile/scrcpy#1966]
- [ ] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [ ] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: [e.g. Debian, Windows, macOS...]
 - scrcpy version: [e.g. 1.12.1]
 - installation method: [e.g. manual build, apt, snap, brew, Windows release...]
 - device model:
 - Android version: [e.g. 10]

**Describe the bug**

On errors, please provide the output of the console (and `adb logcat` if relevant).

```
Please paste terminal output in a code block.
```

Please do not post screenshots of your terminal, just post the content as text instead.

**Comment by rom1v** [comment]
https://github.com/Genymobile/scrcpy/blob/master/FAQ.md#conflicts-between-adb-versions

```bash
ADB=$(which adb) scrcpy
```

**Comment by eater** [comment]
Sorry, github web is being weird but I meant to include that that gives a different error:

```
$ADB=$(which adb) scrcpy                                                                                                                  
INFO: scrcpy 1.16 <https://github.com/Genymobile/scrcpy>
exec: No such file or directory
ERROR: Command not found: [/usr/local/bin/androidtools/platform-tools/adb], [push], [/usr/local/share/scrcpy/scrcpy-server], [/data/local/tmp/scrcpy-server.jar]
ERROR: (make 'adb' accessible from your PATH or define its fullpath in the ADB environment variable)
INFO: You may install 'adb' by "apt install adb"
ERROR: Could not execute "adb push"
```

**Comment by rom1v** [comment]
What is the result of:
```
/usr/local/bin/androidtools/platform-tools/adb devices
```
?

**Comment by eater** [comment]
`List of devices attached`

**Comment by rom1v** [comment]
So for some reason `snap` can't see your host filesystem to use your `adb`.

Try to use a non-snap version (= [build](https://github.com/Genymobile/scrcpy/blob/master/BUILD.md) it).

**Comment by eater** [comment]
Yup, building it works (that's how I've been using scrcpy till now, it's great)

**Comment by Jesus82** [comment]
I'm having this very same problem. 
```
scrcpy
INFO: scrcpy 1.14 <https://github.com/Genymobile/scrcpy>
adb: error: failed to get feature set: no devices/emulators found
ERROR: "adb push" returned with value 1
```
```
sudo scrcpy
INFO: scrcpy 1.14 <https://github.com/Genymobile/scrcpy>
adb server version (41) doesn't match this client (39); killing...
* daemon started successfully
adb: error: failed to get feature set: no devices/emulators found
ERROR: "adb push" returned with value 1
```
```
ADB=$(which adb) scrcpy
INFO: scrcpy 1.14 <https://github.com/Genymobile/scrcpy>
adb server version (39) doesn't match this client (41); killing...
* daemon started successfully
adb: error: failed to get feature set: no devices/emulators found
ERROR: "adb push" returned with value 1
```

```
sudo ADB=$(which adb) scrcpy
INFO: scrcpy 1.14 <https://github.com/Genymobile/scrcpy>
adb: error: failed to get feature set: no devices/emulators found
ERROR: "adb push" returned with value 1
```

Ubuntu 20.10 and Android-Studio downloaded from a repo (not snap) http://ppa.launchpad.net/maarten-fonville/android-studio/ubuntu/

**Comment by fabriziobertoglio1987** [comment]
this issue is caused by two versions of adb installed inthe system.. i think  their package installs the ubuntu apt adb version and saves it in /usr/bin/adb

then we have /home/fabrizio/Android/Sdk/platform-tools/adb and the version from /usr/bin/adb causes this error

I removed such version and error went away

**Comment by ByteBreacher** [comment]
**THIS WILL REALLY HELP YOU**
(almost 90% of the time this might be the problem)
  

1. Download the latest version of ADB from the official Android developer website:

   [https://developer.android.com/studio/releases/platform-tools]

2.  Extract the contents of the ADB ZIP file to a directory on your system, such as ~/Tools.

3.    Add the directory containing the ADB binary to your PATH environment variable by adding the following line to your .bashrc file:
                 
            `export PATH=$PATH:/path/to/adb-file-just-downloaded-and-extracted/`
4. Reload the .bashrc file by running the following command:
           `. ~/.bashrc`


5. Try running `scrcpy` again and see if the error message goes away.
